pkg/sdk/testint/application_packages_integration_test.go

```diff
@@ -0,0 +1,325 @@
+		require.NoError(t, err)
+		assertApplicationPackage(t, id)
+
+		unsetTags := []sdk.ObjectIdentifier{
+			tagTest.ID(),
+		}
+		err = client.ApplicationPackages.Alter(ctx, sdk.NewAlterApplicationPackageRequest(id).WithUnsetTags(unsetTags))
+		require.NoError(t, err)
+		assertApplicationPackage(t, id)
+	})
+
+	t.Run("show application package for SQL: with like", func(t *testing.T) {
+		e := createApplicationPackageHandle(t)
+
+		packages, err := client.ApplicationPackages.Show(ctx, sdk.NewShowApplicationPackageRequest().WithLike(&sdk.Like{Pattern: &e.Name}))
+		require.NoError(t, err)
+		require.Equal(t, 1, len(packages))
+		require.Equal(t, *e, packages[0])
+	})
+}
+
+type StagedFile struct {
+	Name string `json:"name"`
+	Size int    `json:"size"`
+}
+
+type ApplicationPackageVersion struct {
+	Version string `json:"version"`
+	Patch   int    `json:"patch"`
+}
+
+func TestInt_ApplicationPackagesVersionAndReleaseDirective(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	databaseTest, schemaTest := testDb(t), testSchema(t)
+
+	cleanupApplicationPackageHandle := func(id sdk.AccountObjectIdentifier) func() {
+		return func() {
+			err := client.ApplicationPackages.Drop(ctx, sdk.NewDropApplicationPackageRequest(id))
+			if errors.Is(err, sdk.ErrObjectNotExistOrAuthorized) {
+				return
+			}
+			require.NoError(t, err)
+		}
+	}
+
+	createApplicationPackageHandle := func(t *testing.T) *sdk.ApplicationPackage {
+		t.Helper()
+
+		id := sdk.NewAccountObjectIdentifier("snowflake_package_test")
+		request := sdk.NewCreateApplicationPackageRequest(id).WithDistribution(sdk.DistributionPointer(sdk.DistributionInternal))
+		err := client.ApplicationPackages.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupApplicationPackageHandle(id))
+
+		// grant role "ACCOUNTADMIN" on application package
+		_, err = client.ExecForTests(ctx, fmt.Sprintf(`GRANT MANAGE VERSIONS ON APPLICATION PACKAGE "%s" TO ROLE ACCOUNTADMIN;`, id.Name()))
+		require.NoError(t, err)
+
+		e, err := client.ApplicationPackages.ShowByID(ctx, id)
+		require.NoError(t, err)
+		return e
+	}
+
+	createStageHandle := func(t *testing.T) *sdk.Stage {
+		t.Helper()
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, "stage_test")
+		co := sdk.NewStageCopyOptionsRequest().WithOnError(sdk.NewStageCopyOnErrorOptionsRequest().WithSkipFile())
+		cr := sdk.NewCreateInternalStageRequest(id).WithCopyOptions(co)
+		err := client.Stages.CreateInternal(ctx, cr)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err = client.Stages.Drop(ctx, sdk.NewDropStageRequest(id))
+			require.NoError(t, err)
+		})
+
+		e, err := client.Stages.ShowByID(ctx, id)
+		require.NoError(t, err)
+		return e
+	}
+
+	uploadFileForStageHandle := func(t *testing.T, id sdk.SchemaObjectIdentifier, name string) {
+		t.Helper()
+
+		tempFile := fmt.Sprintf("/tmp/%s", name)
+		f, err := os.Create(tempFile)
```

Comment:
you can use https://pkg.go.dev/os#CreateTemp

Reply:
ok, fixed